src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -171,7 +171,7 @@ def __init__(self, cython_call: cygrpc._AioCall, metadata: MetadataType,
                  loop: asyncio.AbstractEventLoop) -> None:
         self._loop = loop
         self._cython_call = cython_call
-        self._metadata = metadata
+        self._metadata = tuple(metadata)
```

Comment:
If we're changing this, we should probably change the annotation on `_metadata` on line 164 to be a bit stronger. Its type should be `Tuple[MetadatumType]` instead of `MetadataType`, which is `Sequence[MetadatumType]`.

Reply:
We still allow `Sequence[MetadatumType]` to pass in, but Cython only recognizes `tuple`. Alternatively, changing all `tuple` to `object` fix the issue (and this is how current stack does it).

Reply:
Understood, but now what I'm referring to. Line 164 is the type annotation for the `_metadata` member, which now has type `Tuple` in all cases due to this new conversion operation.

Reply:
Make sense. I misunderstood the comment. Updated to `Tuple[MetadatumType]` 👍 .

Reply:
I don't get that modification. The ` metdata` parameter is a `MetadataType` type, which is an already sequence of tuples, so here we are adding a new outer tuple, which TBH I do not understand.

So we would end up having something like :

```python
((('foo',bar')))
```

Reply:
@pfreixes `((('foo',bar')))` won't pass the `pytype` check since it is `Tuple[Tuple[MetadatumType]]` instead of `Sequence[MetadatumType]`.

This PR is fixing a misunderstanding on our end that I thought metadata are always `tuple`s. Turned out, both in the document and the type checking, we are allowing `list` metadata. The underlying Cython code is tailored to `tuple`.

As mentioned above, we can potentially fix the Cython to accept `object` instead of `tuple`. However, that means we loses nice features of `tuple`: 1) performance; 2) immutability.

Reply:
My fault, a `tuple(tuple(..))` will produce a `tuple(...)`, since during the constructor the parameter given is iterated

```python
>>> tuple(tuple([(1,2), (2, 3)]))
((1, 2), (2, 3))
```